### PR TITLE
Fix Metaquantome Viz (v2.0.0) Heatmap Clustering

### PR DESCRIPTION
**Pull request by jraysajulga**
### Issue
The **metaQuantome: visualize** (version 2.0.0-0) tool does not run correctly when using the heatmap plotting type.
image

Error message: `Name 'feature_cluster_size' is not defined when running`

_Discovered by @subinamehta_ 

### Reason

An undefined variable (https://github.com/galaxyproteomics/tools-galaxyp/blob/master/tools/metaquantome/metaquantome_viz.xml#L120) from the [758f945](https://github.com/galaxyproteomics/tools-galaxyp/commit/758f9457ee53654a5676aa1c93170a0edd453b35#diff-469d3aad1f38048746ea5c1391fdf0a04d2a845e6bbfdab50a1d8747f1616f05) commit.

### Fix
Prepend '$mode_args.plot' to unknown variables.

#### To do:

- [x] Add/edit tests to define expected heatmap clustering behavior file outputs.

### CODE REVIEW

**Comment by bgruening** [comment]
@jraysajulga would be good to have a test for this option. Thanks.

File: tools/metaquantome/metaquantome_viz.xml [hunk 1]
```diff
@@ -165,16 +165,11 @@
...
-    #if $mode_args.plot.plottype in ["heatmap"]
-        && mv 'feature_cluster_${safename}.txt' '$heatmap_feature_cluster'
-        && mv 'sample_cluster_${safename}.txt' '$heatmap_sample_cluster'
-    #end if
-    #if $mode_args.plot.plottype in ["pca"]
-        && mv 'PC_data_${safename}.txt' '$pca_sample_cluster'
-    #end if
-      
+        </html>" > wrapped_outfile.html &&
+        if test -f 'feature_cluster_${infile.name}.txt'; then cp 'feature_cluster_${infile.name}.txt' feature_cluster_data.txt; fi &&
```
Comment (bgruening) [inline tools/metaquantome/metaquantome_viz.xml:169]: mv instead of cp?

File: tools/metaquantome/metaquantome_viz.xml [hunk 2]
```diff
@@ -389,15 +387,25 @@
...
-        <test expect_num_outputs="3">
-             <param name="infile" value="tax_test_out.tab" ftype="tabular"/>
-             <param name="mode" value="t" />
-             <param name="samps" value="rudney_samples.tab" ftype="tabular"/>
-             <param name="plottype" value="heatmap" />
-             <output name="wrapped_outfile">
-                <assert_contents>
-                    <has_text text="To download the image" />
-                </assert_contents>
+        <test>
```
Comment (bgruening) [inline tools/metaquantome/metaquantome_viz.xml:390]: can you please add `expect_num_outputs` to the tests. See here: https://docs.galaxyproject.org/en/latest/dev/schema.html#id80

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
changes looks good, but tests are failing infortunately :(

**Comment by jraysajulga** [comment]
Okay. Do you have any tips for using `planemo`? I'm trying to `planemo test` and `planemo serve` but they both get stuck while running the tool.

**Comment by jraysajulga** [comment]
Which tests are failing?
image

**Comment by bgruening** [comment]
Go here: https://github.com/galaxyproteomics/tools-galaxyp/pull/504/checks?check_run_id=1766314904

Click on "artifacts" and download the "All tool tests results"

**Comment by jraysajulga** [comment]
Okay. Thanks! That's quite helpful. Seems like the last error is during the lint:
```
.. ERROR: Invalid XML found in file: tmps78h8uak. Errors [/tmp/tmps78h8uak:398:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_3_2_1: Element 'test', attribute 'expect_num_output': The attribute 'expect_num_output' is not allowed.]
```

**Comment by bgruening** [comment]
expect_num_outputs plese see here: https://docs.galaxyproject.org/en/latest/dev/schema.html#id80

**Review by bgruening — APPROVED** [review]

**Pull request merged**